src/Microsoft.Health.Dicom.Api/Features/Converter/CustomJsonStringEnumConverterImp.cs

```diff
@@ -0,0 +1,40 @@
+﻿// -------------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License (MIT). See LICENSE in the repo root for license information.
+// -------------------------------------------------------------------------------------------------
+
+using System;
+using System.Globalization;
+using System.Text.Json;
+using System.Text.Json.Serialization;
+using EnsureThat;
+
+namespace Microsoft.Health.Dicom.Api.Features.Converter
+{
+    internal class CustomJsonStringEnumConverterImp<T> : JsonConverter<T>
+        where T : Enum
+    {
+        public override T Read(ref Utf8JsonReader reader, Type typeToConvert, JsonSerializerOptions options)
+        {
+            EnsureArg.IsNotNull(typeToConvert, nameof(typeToConvert));
+
+            if (reader.TokenType == JsonTokenType.String)
+            {
+                string content = reader.GetString();
+                if (Enum.TryParse(typeToConvert, content, true, out object result))
+                {
+                    return (T)result;
+                }
+            }
+
+            throw new JsonException(
+                string.Format(CultureInfo.InvariantCulture, DicomApiResource.InvalidEnumValue, string.Join("\",\"", Enum.GetNames(typeToConvert))));
```

Comment:
While helpful, I am unsure we always want to print all of the possible values in the error. When things like `Enum.Parse` fail, they simply say they cannot parse it. Perhaps that is sufficient, especially if we ever have very prolific enums. Users can always refer to the swagger

Reply:
Another approach is to create custom jsonconverter for these enums specifically.